x-pack/plugins/transform/public/alerting/transform_health_rule_type/transform_selector_control.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBox, EuiComboBoxProps, EuiFormRow } from '@elastic/eui';
+import React, { FC, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ALL_TRANSFORMS_SELECTION } from '../../../common/constants';
+import { isDefined } from '../../../common/types/common';
+
+export interface TransformSelectorControlProps {
+  label?: string | JSX.Element;
+  errors?: string[];
+  onChange: (transformSelection: string[]) => void;
+  selectedOptions: string[];
+  options: string[];
+  allowSelectAll?: boolean;
+}
+
+function convertToEuiOptions(values: string[]) {
+  return values.map((v) => ({ value: v, label: v }));
+}
+
+export const TransformSelectorControl: FC<TransformSelectorControlProps> = ({
+  label,
+  errors,
+  onChange,
+  selectedOptions,
+  options,
+  allowSelectAll = false,
+}) => {
+  const onSelectionChange: EuiComboBoxProps<string>['onChange'] = ((selectionUpdate) => {
+    if (!selectionUpdate?.length) {
+      onChange([]);
+      return;
+    }
+    if (selectionUpdate[selectionUpdate.length - 1].value === ALL_TRANSFORMS_SELECTION) {
+      onChange([ALL_TRANSFORMS_SELECTION]);
+      return;
+    }
+    onChange(
+      selectionUpdate
+        .slice(selectionUpdate[0].value === ALL_TRANSFORMS_SELECTION ? 1 : 0)
+        .map((v) => v.value)
+        .filter(isDefined)
+    );
+  }) as Exclude<EuiComboBoxProps<string>['onChange'], undefined>;
+
+  const selectedOptionsEui = useMemo(() => convertToEuiOptions(selectedOptions), [selectedOptions]);
+  const optionsEui = useMemo(() => {
+    return convertToEuiOptions(allowSelectAll ? [ALL_TRANSFORMS_SELECTION, ...options] : options);
+  }, [options, allowSelectAll]);
+
+  return (
+    <EuiFormRow
+      fullWidth
+      label={
+        label ?? (
+          <FormattedMessage
+            id="xpack.transform.transformSelector.formControlLabel"
```

Comment:
This needs a better ID.

Reply:
Actually, this one is redundant. Removed in https://github.com/elastic/kibana/pull/112277/commits/6c27a241a1860d001eb6bcd84fec2cb8ec504b0c